documentation/user-consent.md

```diff
@@ -0,0 +1,214 @@
+# User Consent Management
+
+This document captures guidelines for Operator NaaS platform to handle user consents.
+
+## Table of Contents
+
+- [User Consent Management](#user-consent-management)
+  - [Table of Contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [Glossary of Terms and Concepts](#glossary-of-terms-and-concepts)
+  - [Purposes](#purposes)
+    - [Purpose levels](#purpose-levels)
+    - [Purpose relationship with other key concepts](#purpose-relationship-with-other-key-concepts)
+    - [Purpose definition](#purpose-definition)
+    - [Using purpose parameter in the authorization request](#using-purpose-parameter-in-the-authorization-request)
+  - [User Identity](#user-identity)
+  - [User Authentication/Authorization \& Consent Management](#user-authenticationauthorization--consent-management)
+    - [Authentication mechanism/s](#authentication-mechanisms)
+    - [Consent capture and storage](#consent-capture-and-storage)
+    - [Athorization flows / grant types](#athorization-flows--grant-types)
+      - [Authorization code flow](#authorization-code-flow)
+      - [JWT-bearer flow](#jwt-bearer-flow)
+      - [CIBA flow](#ciba-flow)
+      - [Client Credentials flow](#client-credentials-flow)
+      - [... (other grant types)](#-other-grant-types)
+  - [Consent API](#consent-api)
+  - [Annex: Flow source code](#annex-flow-source-code)
+
+
+## Introduction
+
+Operator NaaS platform implementing CAMARA should be built with a privacy-by-design approach to fully comply with data protection regulations like the [GDPR regulation](https://gdpr-info.eu/) in Europe, which provides a great level of protection for user's privacy. This means that an API that processes Personal Information needs a user consent. Consents are given by users to legal entities to process personal data under a specific purpose.
+
+This document captures guidelines for Operator NaaS platform to handle user consents to comply with GDPR or equivalent requirements in an easy way, introducing the concept of purpose in OpenID Connect. Even being defined based on concepts that maps to GDPR regulation, proposed solution and concepts are generic enough to be used by Operators on any country.
+
+>[TO BE EDITED/COMPLETED]
+
+## Glossary of Terms and Concepts
+
+The list below introduces several key concepts:
+
+-	`NaaS`: Network-as-a-Service model where Telco Network capabilities are exposed to third parties through APIs. And some of those APIs are likely to require user consent to be accessed. 
```

Comment:
Submitted an issue already for this https://github.com/camaraproject/IdentityAndConsentManagement/issues/17

Reply:
Fine, I removed it in this commit https://github.com/camaraproject/IdentityAndConsentManagement/pull/13/commits/150a596f77633aa771dbadbebf6bc00b107a5762